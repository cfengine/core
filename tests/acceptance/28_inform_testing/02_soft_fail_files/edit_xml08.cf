body file control
{
  inputs => { "$(sys.policy_entry_dirname)/../common_soft_fail.cf.sub" };
}

bundle common testcase
{
  vars:
    "filename" string => "$(this.promise_filename)";
}


bundle agent setup
{

  files:
   "$(G.testfile)" edit_xml => xml_insert_tree_nopath("<Root/>"),
      create => "true";
}

bundle agent main
{
  files:
    "$(G.testfile)" edit_xml => xml_insert_text("Tuesday", "//Root");
}

bundle agent teardown
{
  reports:
    "Edited file has contents: "
      printfile => my_cat( $(G.testfile) );
}

body printfile my_cat(file)
# @brief Report the contents of a file
# @param file The full path of the file to report
{
        file_to_print => "$(file)";
        number_of_lines => "inf";
}

bundle edit_xml xml_insert_tree_nopath(treestring)
{
  insert_tree:
      '$(treestring)';
}

bundle edit_xml xml_insert_text(text, xpath)
{
  insert_text:
    "$(text)"
      select_xpath => "$(xpath)";
}
