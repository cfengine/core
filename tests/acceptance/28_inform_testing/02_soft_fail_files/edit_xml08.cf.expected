    info: files promise '/tmp/TEST.cfengine' repaired, edit_xml => xml_insert_text("Tuesday", "//Root")
    info: Inserted text 'Tuesday' at XPath '//Root' in the XML document '/tmp/TEST.cfengine'
